cmd/xl-storage.go

```diff
@@ -131,13 +131,14 @@ type xlStorage struct {
 // checkPathLength - returns error if given path name length more than 255
 func checkPathLength(pathName string) error {
 	// Apple OS X path length is limited to 1016
-	if runtime.GOOS == "darwin" && len(pathName) > 1016 {
+	if runtime.GOOS == globalMacOSName && len(pathName) > 1016 {
 		return errFileNameTooLong
 	}
 
-	// Disallow more than 1024 characters on windows, there
+	// Disallow more than 32,767 characters on windows, there
 	// are no known name_max limits on Windows.
-	if runtime.GOOS == "windows" && len(pathName) > 1024 {
+	// Refer: https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry
+	if runtime.GOOS == globalWindowsOSName && len(pathName) > 32767 {
```

Comment:
this is not suitable for MinIO - we need to return error for object names > 1024 so this is not the same situation as `mc`

Reply:
correct but this path is well with in that margin, we don't care that windows supports 32k characters.